texlab won't build on macOS

**Describe the bug**
When I try to install `texlab` on macOS 10.15 with `nix-darwin`, I got the following error:
```
...............] \ : info lifecycle core-js@2.6.11~postinstall: core-js@2.6[0m
> fsevents@1.2.11 install /nix/store/722inglv8pcrz2wgal76rg47zp2153mz-node_citeproc-0.1.0/lib/node_modules/citeproc/node_modules/fsevents
> node-gyp rebuild

make: Entering directory '/nix/store/722inglv8pcrz2wgal76rg47zp2153mz-node_citeproc-0.1.0/lib/node_modules/citeproc/node_modules/fsevents/build'
  SOLINK_MODULE(target) Release/.node
  CXX(target) Release/obj.target/fse/fsevents.o
../fsevents.cc:10:10: fatal error: 'CoreServices/CoreServices.h' file not found
#include "CoreServices/CoreServices.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [fse.target.mk:123: Release/obj.target/fse/fsevents.o] Error 1
make: Leaving directory '/nix/store/722inglv8pcrz2wgal76rg47zp2153mz-node_citeproc-0.1.0/lib/node_modules/citeproc/node_modules/fsevents/build'

```

**To Reproduce**
Steps to reproduce the behavior:
1. List `texlab` in `environment.systemPackages`
2. `darwin-rebuild switch`

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.2.0, macOS 10.15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(syp): `""`
 - channels(root): `"darwin, nixpkgs-20.03pre206486.a8fc904c7c0"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/node-packages/default-v10.nix
+++ b/pkgs/development/node-packages/default-v10.nix
@@ -95,6 +95,10 @@ nodePackages // {
     '';
   };
 
+  texlab-citeproc-build-deps = nodePackages."texlab-citeproc-build-deps-../tools/misc/texlab/citeproc".override {
+    buildInputs = stdenv.lib.optionals stdenv.isDarwin [ pkgs.darwin.apple_sdk.frameworks.CoreServices ];
+  };
+
   webtorrent-cli = nodePackages.webtorrent-cli.override {
     buildInputs = [ nodePackages.node-gyp-build ];
   };

--- a/pkgs/development/node-packages/node-packages-v10.json
+++ b/pkgs/development/node-packages/node-packages-v10.json
@@ -15,7 +15,6 @@
 , "bower2nix"
 , "browserify"
 , "castnow"
-, { "texlab-citeproc-build-deps": "../tools/misc/texlab/citeproc" }
 , "clean-css"
 , "coffee-script"
 , "coinmon"
@@ -126,6 +125,7 @@
 , "swagger"
 , {"tedicross": "git+https://github.com/TediCross/TediCross.git#v0.8.7"}
 , "tern"
+, { "texlab-citeproc-build-deps": "../tools/misc/texlab/citeproc" }
 , "textlint"
 , "textlint-plugin-latex"
 , "textlint-rule-abbr-within-parentheses"

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -10589,7 +10589,7 @@ in
 
   texlab = callPackage ../development/tools/misc/texlab {
     inherit (darwin.apple_sdk.frameworks) Security;
-    texlab-citeproc-build-deps = nodePackages."texlab-citeproc-build-deps-../tools/misc/texlab/citeproc";
+    inherit (nodePackages) texlab-citeproc-build-deps;
   };
 
   tflint = callPackage ../development/tools/analysis/tflint { };

